### PR TITLE
Update book 1 renders plus many listing fixes

### LINKED ISSUE #547: Update render 1.05
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #548: Update render 1.06
- Render as a portion of image 1.05 (let Markdeep blow it up)
- PNG format

Ref #179

### LINKED ISSUE #549: Update render 1.07
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #550: Update render 1.08
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #551: Update render 1.02
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #552: Update render 1.03
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #553: Update render 1.04
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #554: Update render 1.09
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #555: Update render 1.10
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #556: Update render 1.11
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #557: Update render 1.12
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #560: Update render 1.15
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #561: Update render 1.16
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #562: Update render 1.17
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #563: Update render 1.18
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #564: Update render 1.19
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #565: Update render 1.20
- Width 400 (× 225)
- PNG format

Ref #179

### LINKED ISSUE #566: Update render 1.21
- Width 400 (× 225)
- PNG format

Ref #179

### PR DESCRIPTION
**Pull request by hollasch**
- Reduced code duplication in dielectric::scatter()
- Change standard render dimensions to 400x225
- Change image 6 to a before-and-after pair to show antialiasing
- Refactored material and geometry declarations (multiple listings)
- Refactored assignment of `etai_over_etat`
- Fix multiple listing highlighting misses
- Fix multiple listing include errors
- Add missing `double t` member of struct `hit_record`
- Add missing scene code changes at various points
- Show final scene render parameters plus highlighting

See #179
Resolves #547
Resolves #548
Resolves #549
Resolves #550
Resolves #551
Resolves #552
Resolves #553
Resolves #554
Resolves #555
Resolves #556
Resolves #557
Resolves #560
Resolves #561
Resolves #562
Resolves #563
Resolves #564
Resolves #565
Resolves #566

### CODE REVIEW

**Comment by trevordblack** [comment]
Really like the before/after shot with antialiasing

File: books/RayTracingInOneWeekend.html [hunk 1]
```diff
@@ -2364,11 +2405,10 @@
 solution does not exist, the glass cannot refract, and therefore must reflect the ray:
```
Comment (trevordblack) [inline books/RayTracingInOneWeekend.html:2405]: Trailing comma on line 2401

**Comment by trevordblack** [comment]
I'm seeing really weird artifacting on some of the material renders.
See:
img-1.11-metal-shiny.png
img-1.12-metal-fuzz.png
img-1.15-glass-sometimes-refract.png
img-1.16-glass-hollow.png
img-1.18-view-distant.png
img-1.19-view-zoom.png
img-1.21-book1-final.png

We need to figure out where those are coming from

**Comment by trevordblack** [comment]
So, I'm rerendering the images and i'm not getting artifacts, I'll have them up as a commit in this branch.
I'm also fixing:
trailing comma (see above)
bubble thickness

**Comment by trevordblack** [comment]
I'm wrapping a bunch of small changes into my commit

**Comment by hollasch** [comment]
Why would your images look different? Like you say, we should figure this out.

Also, if we end up with better images, we'll need to construct a pristine pull request. I don't want to bloat the repo database with the unused images.

**Comment by hollasch** [comment]
I presume you're using the `progression` branch to do the renders, yes?

**Comment by trevordblack** [comment]
I'm not sure why my images are different. I'm running ubuntu w clang9.

I'm actually rerendering almost all of the images

**Comment by trevordblack** [comment]
Nope. I'm adding directly to `render-updates`.
Mistake?

**Comment by hollasch** [comment]
I'll retry with the modern C++ RNG — there's no good reason for us to use the old one.

**Comment by hollasch** [comment]
Yes — use the `progression` branch. This will also ensure that you're not accidentally slipstreaming in any future or personal modifications, and are following the progression as the book presents.

**Comment by trevordblack** [comment]
Okay, go ahead and try. BUT, I don't think your artifacting is the RNG

**Comment by trevordblack** [comment]
ahhh, shucks
I'll move over to progression

**Comment by trevordblack** [comment]
Okay, I'm seeing now that `progression`/`dev-minor`/`render-updates` is a rats nest.
I'm going to create a branch off of `progression` that has ONLY the image updates, and create a PR.
I'm rendering the final scene rn, which could take another hour.

Everything is rendered at 400x225 (100spp), except for the final scene, which is 1200x800 (500spp)

**Comment by hollasch** [comment]
The book stipulates 50spp. The intent is to enable fast turnaround while developing, not good-looking pictures.

**Comment by trevordblack** [comment]
Wait. What should I be rendering at?

**Comment by hollasch** [comment]
400×225, 50spp. But all of this is coded/written in the progression branch. So you just need to visit the individual commits and run. For example, `git checkout progression~20` to get to the codebase reset. `progression~19` is img-1.01, `progression~18` is img-1.02, `progression~17` is img-1.03, and so on.

**Comment by hollasch** [comment]
@trevordblack helped me figure out that I'd missed the surface-acne epsilon, which caused all sorts of subtle rendering artifacts. Walking through the progression uncovered more small issues. Going to wad this one and recreate for a second pass.

**Pull request closed without merging**